tests: kernel: mem_slab: mslab_api: undefined reference to z_impl_k_sem_give and z_impl_k_sem_take

**Describe the bug**
Builds are failing in CI for `qemu_cortex_m0` and the test case `tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading` because `k_sem_give()` and `k_sem_take()` have no implementation.

Seems to work fine on other platforms.

**To Reproduce**
```
twister -i -p qemu_cortex_m0 -s tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading
```

**Expected behavior**
CI should be running smoothly.

**Impact**
Showstopper

**Logs and console output**
```
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(test_mslab_api.c.obj): in function `k_sem_take':
/workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1026: undefined reference to `z_impl_k_sem_take'
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(test_mslab_api.c.obj): in function `k_sem_give':
/workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1043: undefined reference to `z_impl_k_sem_give'
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1043: undefined reference to `z_impl_k_sem_give'
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1043: undefined reference to `z_impl_k_sem_give'
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1043: undefined reference to `z_impl_k_sem_give'
/opt/toolchains/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /workdir/zephyr/twister-out/qemu_cortex_m0/tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api_no_multithreading/zephyr/include/generated/syscalls/kernel.h:1043: undefined reference to `z_impl_k_sem_give'
```
**Environment (please complete the following information):**
 - OS: CI
 - Toolchain: Zephyr SDK v0.13.1
 - Commit: e8f09b471e

**Additional context**
https://buildkite.com/zephyr/zephyr/builds/45153

I have not narrowed down the exact commit that broke this test.

--- a/tests/kernel/mem_slab/mslab_api/src/test_mslab_api.c
+++ b/tests/kernel/mem_slab/mslab_api/src/test_mslab_api.c
@@ -133,10 +133,16 @@ static void tmslab_used_get(void *data)
 
 K_SEM_DEFINE(SEM_HELPERDONE, 0, 1);
 K_SEM_DEFINE(SEM_REGRESSDONE, 0, 1);
+static K_THREAD_STACK_DEFINE(stack, STACKSIZE);
+static struct k_thread HELPER;
 
-static void helper_thread(void)
+static void helper_thread(void *p0, void *p1, void *p2)
 {
 	void *ptr[BLK_NUM];           /* Pointer to memory block */
+
+	ARG_UNUSED(p0);
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
 
 	(void)memset(ptr, 0, sizeof(ptr));
 
@@ -169,9 +175,6 @@ static void helper_thread(void)
 	k_sem_give(&SEM_HELPERDONE);
 
 }  /* helper thread */
-
-K_THREAD_DEFINE(HELPER, STACKSIZE, helper_thread, NULL, NULL, NULL,
-		7, 0, 0);
 
 /*test cases*/
 /**
@@ -289,8 +292,17 @@ void test_mslab_used_get(void)
  */
 void test_mslab_pending(void)
 {
+	if (!IS_ENABLED(CONFIG_MULTITHREADING)) {
+		ztest_test_skip();
+		return;
+	}
+
 	int ret_value;
 	void *b;                        /* Pointer to memory block */
+
+	(void)k_thread_create(&HELPER, stack, STACKSIZE,
+			helper_thread, NULL, NULL, NULL,
+			7, 0, K_NO_WAIT);
 
 	k_sem_give(&SEM_REGRESSDONE);   /* Allow helper thread to run */
 

